### PR TITLE
feat: error handling for executor timeouts

### PR DESCRIPTION
**Pull request by JohannesMessner**
**Goal:** Handle timeouts the same way that connection errors are handled: retries are applied, and useful error messages are propagated to the user.

**ToDo:**
- [x] gRPC gateway
- [x] http gateway
- [x] ws gateay
- [x] testing
- [x] docs

### CODE REVIEW

File: jina/serve/runtimes/gateway/graph/topology_graph.py [hunk 1]
```diff
@@ -65,7 +65,15 @@ def _handle_internalnetworkerror(self, err):
...
-                    + f' |Gateway: Communication error with deployment {self.name} at address(es) {err.dest_addr}. Head or worker(s) may be down.'
+                    + f' |Gateway: Communication error with deployment {self.name} at address(es) {err.dest_addr}. '
+                    f'Head or worker(s) may be down.'
+                )
+                raise err
+            elif err_code == grpc.StatusCode.DEADLINE_EXCEEDED:
+                err._details = (
+                    err.details()
+                    + f'|Gateway: Connection with deployment {self.name} at address(es) {err.dest_addr} could be established, but timed out.'
+                    f' You can increase the allowed time by setting `Flow(timeout_send=time_in_ms)`.'
```
Comment (JoanFM) [inline jina/serve/runtimes/gateway/graph/topology_graph.py:76]: maybe change to setting `timeout_send` attribute in your Flow, which can aplly also to a Flow with `yaml`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/gateway_clients/test_executor_timeout_failures.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    p.start()
+    p.join()
+    assert (
+        p.exitcode == 0
+    )  # if exitcode != 0 then test in other process did not pass and this should fail
+
+
+@pytest.mark.parametrize('protocol', ['grpc', 'http', 'websocket'])
+def test_headfull_exec_timeout(port_generator, protocol):
+    f = Flow(timeout_send=1, protocol=protocol).add(uses=SlowExecutor, shards=2)
```
[the remark this answers is not in the document]
Comment (JohannesMessner) [reply tests/integration/gateway_clients/test_executor_timeout_failures.py:39]: it is passed to the `_test_error` method in a new process and enters a context manager there. Unfortunately gRPC and pytest don't play nice together, so this has to be done in a new subprocess. I don't think entering the Flow context manager here would work.

[the remark this answers is not in the document]
Comment (JoanFM) [reply tests/integration/gateway_clients/test_executor_timeout_failures.py:39]: please do not pass a Flow to another process, it is a bad pattern, pass perhaps the arguments but not the Flow, if it is already like this in other tests please change.

Pickling a Flow object is a little weird

[the remark this answers is not in the document]
Comment (JohannesMessner) [reply tests/integration/gateway_clients/test_executor_timeout_failures.py:39]: Ok that makes sense, didn't think about it too much. Will change

**Review by JoanFM — APPROVED** [review]

**Pull request merged**